scripts/leapp_preupgrade.py

```diff
@@ -0,0 +1,340 @@
+        report_entries = report_json.get("entries", [])
+        inhibitor_count = len(
+            [entry for entry in report_entries if "inhibitor" in entry.get("flags")]
+        )
+        message = "Your system has %s inhibitors out of %s potential problems." % (
+            inhibitor_count,
+            len(report_entries),
+        )
+        alert = inhibitor_count > 0
+
+    output.report_json = report_json
+    output.alert = alert
+    output.message = message
+
+    print("Reading TXT report")
+    text_report_path = "/var/log/leapp/leapp-report.txt"
+    report_txt = "Not found"
+    if os.path.exists(text_report_path):
+        with open(json_report_path, mode="r") as handler:
+            report_txt = handler.read()
+
+    output.report = report_txt
+
+
+def call_insights_client():
+    print("Calling insight-client in background for immediate data collection.")
+    run_subprocess(["insights-client"], wait=False)
+    # NOTE: we do not care about returncode or output because we are not waiting for process to finish
+
+
+def main():
+    # Exit if not RHEL 7 or 8
+    dist, version = get_rhel_version()
+    if dist != "rhel" or exit_for_non_eligible_releases(version):
+        raise ProcessError(
+            message='Exiting because distribution="%s" and version="%s"'
+            % (dist, version)
+        )
+
+    output = OutputCollector()
+
+    try:
+        # Init variables
+        preupgrade_command = ["/usr/bin/leapp", "preupgrade", "--report-schema=1.1.0"]
+        use_no_rhsm = False
+        rhui_pkgs = []
+        if version.startswith("7"):
+            leapp_install_command = [
+                "yum",
+                "install",
+                "leapp-upgrade",
+                "-y",
+                "--enablerepo=rhel-7-server-extras-rpms",
+            ]
+            rhel_7_rhui_packages = [
+                {"src_pkg": "rh-amazon-rhui-client", "leapp_pkg": "leapp-rhui-aws"},
+                {
+                    "src_pkg": "rh-amazon-rhui-client-sap-bundle",
+                    "leapp_pkg": "leapp-rhui-aws-sap-e4s",
+                },
+                {"src_pkg": "rhui-azure-rhel7", "leapp_pkg": "leapp-rhui-azure"},
+                {
+                    "src_pkg": "rhui-azure-rhel7-base-sap-apps",
+                    "leapp_pkg": "leapp-rhui-azure-sap",
+                },
+                {
+                    "src_pkg": "rhui-azure-rhel7-base-sap-ha",
+                    "leapp_pkg": "leapp-rhui-azure-sap",
+                },
+                {
+                    "src_pkg": "google-rhui-client-rhel7",
+                    "leapp_pkg": "leapp-rhui-google",
+                },
+                {
+                    "src_pkg": "google-rhui-client-rhel79-sap",
+                    "leapp_pkg": "leapp-rhui-google-sap",
+                },
+            ]
+            rhui_pkgs = setup_leapp(leapp_install_command, rhel_7_rhui_packages)
+        if version.startswith("8"):
```

Comment:
We don't have the worker prepared to execute it under RHEL8 at the moment, but maybe the idea here from @andywaltlova was to prepare the ground for the future?

Reply:
I know we don't, but I think we should have both yamls with same funcionality for preupgrade, I think it could come in handy in future. we could decide to drop it after we know how insights will show two different preupgrade tasks. (we could then also drop rhel7 stuff from the ansible)

Reply:
Sure we can keep it as of now. But if it will only be used on rhel7, I would rather delete it in future.